HolyshiftService/PasswordHashService.cs

```diff
@@ -0,0 +1,30 @@
+using Konscious.Security.Cryptography;
+using System;
+using System.Text;
+
+public class PasswordHashService
+{
+    public string GenerateRandomSalt(int size)
+    {
+        byte[] salt = new byte[size];
+        using (var rng = System.Security.Cryptography.RandomNumberGenerator.Create())
+            rng.GetBytes(salt);
+
+        return Convert.ToBase64String(salt);
+    }
+
+    public async Task<string> HashPassword(string password, string saltBase64)
+    {
+        byte[] salt = Convert.FromBase64String(saltBase64);
+
+        using var hasher = new Argon2id(Encoding.UTF8.GetBytes(password));
+        hasher.Salt = salt;
+        hasher.DegreeOfParallelism = 1;
+        hasher.MemorySize = 9216;
```

Comment:
Let's write this as `9 * 1024` for readability.
And add a comment next to it `9 MiB`